test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSourcesCommandTest.cs

```diff
@@ -42,6 +42,337 @@ public void SourcesCommandTest_AddSource()
+                    "-ConfigFile",
+                    settings.ConfigPath
+                };
+
+                // Act
+                var result = CommandRunner.Run(nugetexe, workingPath, string.Join(" ", args), true);
+
+                // Assert
+                Assert.Equal(0, result.ExitCode);
+                var loadedSettings = Configuration.Settings.LoadDefaultSettings(workingPath, null, null);
+                var packageSourcesSection = loadedSettings.GetSection("packageSources");
+                var sourceItem = packageSourcesSection?.GetFirstItemWithAttribute<SourceItem>("key", "test_source");
+                Assert.Equal("https://test_source", sourceItem.GetValueAsPath());
+                Assert.False(result.Output.Contains("WARNING"));
+            }
+        }
+
+        [Fact]
+        public void SourcesCommandTest_UpdateSource_WarnWhenUsingHttp()
+        {
+            using (var configFileDirectory = TestDirectory.Create())
+            {
+                var nugetexe = Util.GetNuGetExePath();
+                var configFileName = "nuget.config";
+                var configFilePath = Path.Combine(configFileDirectory, configFileName);
+
+                var nugetConfig =
+                    @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <add key=""test_source"" value=""http://source.test"" />
+  </packageSources>
+</configuration>";
+                Util.CreateFile(configFileDirectory, configFileName, nugetConfig);
+
+                // Arrange
+                var args = new string[] {
+                    "sources",
+                    "Update",
+                    "-Name",
+                    "test_source",
+                    "-Source",
+                    "http://test_source",
+                    "-ConfigFile",
+                    configFilePath
+                };
+
+                // Act
+                var result = CommandRunner.Run(
+                    nugetexe,
+                    configFileDirectory,
+                    string.Join(" ", args),
+                    true);
+
+                // Assert
+                Assert.Equal(0, result.ExitCode);
+                var loadedSettings = Configuration.Settings.LoadDefaultSettings(configFileDirectory, configFileName, null);
+                var packageSourcesSection = loadedSettings.GetSection("packageSources");
+                var sourceItem = packageSourcesSection?.GetFirstItemWithAttribute<SourceItem>("key", "test_source");
+                Assert.Equal("http://test_source", sourceItem.GetValueAsPath());
+                Assert.True(result.Output.Contains("WARNING: You are running the 'update source' operation with an 'HTTP' source, 'http://test_source'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source."));
+            }
+        }
+
+        [Fact]
+        public void SourcesCommandTest_UpdateSource_NoWarnWhenUsingHttps()
```

Comment:
Same here. I guess it's a philosophical decision (somewhat nitpicking), but I prefer 1 test with parameters. Lots of duplicate code just to have a true assertion and a false assertion.